apps/src/templates/hintsDisplay.jsx

```diff
@@ -0,0 +1,57 @@
+module.exports = React.createClass({
+
+  propTypes: {
+    hintReviewTitle: React.PropTypes.string.isRequired,
+    getSeenHints: React.PropTypes.func.isRequired,
+    getUnseenHints: React.PropTypes.func.isRequired,
+    recordUserViewedHint: React.PropTypes.func.isRequired,
+    lightbulbSVG: React.PropTypes.node.isRequired,
+  },
+
+  getInitialState: function () {
+    return {
+      seenHints: this.props.getSeenHints(),
+      unseenHints: this.props.getUnseenHints(),
+      showViewHintButton: true
+    };
+  },
+
+  viewHint: function () {
+    this.props.recordUserViewedHint(this.state.unseenHints[0]);
+    var newState = this.getInitialState();
```

Comment:
In general, the React philosophy is that state is bad and makes things harder to reason about and to test. In a world without state, I can look at the incoming props and figure out exactly what the component will look like. In a world where these are state, I now need to understand a bunch more about how getSeenHints and getUnseenHints work